File: day3/main.js
const fs = require('fs');

const scores = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

const input = fs.readFileSync('in.txt', "utf-8").split("\r\n");

let score = 0;
for (const rs of input) {
    const size = rs.length;
    const c1 = rs.slice(0, size / 2);
    const c2 = rs.slice(size / 2);

    for (const item of c2) {
        if (c1.includes(item)) {
            score += scores.indexOf(item) +1
            break
        }
    }
}

console.log(score)


score = 0
for (let i = 0; i < input.length; i+=3) {
    const [r1, r2, r3] = input.slice(i, i+3);
    for (const item of r1) {
        if (r2.includes(item) && r3.includes(item)) {
            score += scores.indexOf(item) +1
            break
        }
    }
}
console.log(score)

File: day3/main.py
scores = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

with open("in.txt", "r") as f:
    score = 0
    for rs in f:
        size = len(rs)
        c1 = rs[:size//2]
        c2 = rs[size//2:]
        for item in c2:
            if (item in c1):
                score += scores.index(item) +1
                break

    print(score)


with open("in.txt", "r") as f:
    input = f.read().splitlines()
    score = 0
    for i in range(0, len(input), 3):
        r1, r2, r3 = input[i:i+3]
        for item in r1:
            if (item in r2 and item in r3):
                score += scores.index(item) +1
                break

    print(score)